src/Microsoft.AspNet.OData.Shared/Formatter/ODataModelBinderConverter.cs

```diff
@@ -30,8 +30,20 @@ namespace Microsoft.AspNet.OData.Formatter
         Justification = "Relies on many ODataLib classes.")]
     public static class ODataModelBinderConverter
     {
-        private static readonly MethodInfo EnumTryParseMethod = typeof(Enum).GetMethods()
+#if NETSTANDARD2_0
```

Comment:
I think you should have he `#if` macro checking for new versions. The solution is currently failing to build.

So maybe `#if NETSTANDARD2_1` or something specific to NET 5.x and above. Then the "legacy" implementation would remain in the `else` block.

Reply:
Web API OData is not supporting anything older than .NET Standard 2.0
So we've got the following table:
1- .NET frameworks older than 4.6.1 is not supported at all
2- .NET framework 4.6.1+ (Up to .NET framework 4.8) will choose #if NETSTANDARD2_0 part automatically.
3- .NET Core older than .net core 2.0 are not supported at all.
4- .NET Core 2.0 up to .NET Core 3.0 will choose #if NETSTANDARD2_0 part automatically.
5- .NET Core 3.1 will choose #else part
6- .NET 5 and 6 and upcoming versions will choose #else part.

I think everything is alright unless I'm getting something wrong

Reply:
Oh, I see the build error. Let me check